pinot-query-planner/src/main/java/org/apache/pinot/query/parser/utils/ParserUtils.java

```diff
@@ -53,95 +54,87 @@ public static boolean canCompileWithMultiStageEngine(String query, String databa
   }
 
   /**
-   * Tries to fill an empty or not properly filled schema when no rows have been returned.
+   * Tries to fill an empty or not properly filled {@link DataSchema} when no row has been returned.
+   *
+   * Response data schema can be inaccurate or incomplete in several forms:
+   * 1. No result table at all (when all segments have been pruned on broker).
+   * 2. Data schema has all columns set to default type (STRING) (when all segments pruned on server).
```

Comment:
I have contradictory feelings about explaining why some issues can happen in the generic method that solves them. On the one hand, it is useful to have the reasons centralized. On the other hand, it is a clear candidate for documentation that could be desynchronized with the code (ie we change one of these cases and forget to change the javadoc).

Reply:
We probably need a dev focus documentation to document the internal behaviors. It is always a hard problem to keep the doc in sync with code though. I'd prefer having it in javadoc for now for future developers to know the history